backend/app/store/service/service.go

```diff
@@ -796,6 +797,34 @@ func (s *DataStore) Delete(locator store.Locator, commentID string, mode store.D
 		s.repliesCache.Delete(commentID)
 		s.repliesCache.Delete(comment.ParentID)
 	}
+
+	// delete images from the comment if they are not reused elsewhere in comments to the same page
```

Comment:
could be my misunderstanding but why we even care about the "if they are not reused elsewhere" part. Isn't is is safe to assume the image is unique to the particular comment and not bother with all those checks? You likely had some use-case (potential collision) in mind, but I fail to see it

Reply:
From the top of my mind, I can quote your message, and then the same image would be used in two places. Deleting the quoting message is not supposed to delete the original comment message.